cmd/containerd/command/main.go

```diff
@@ -274,4 +274,13 @@ func dumpStacks() {
 	}
 	buf = buf[:stackSize]
 	logrus.Infof("=== BEGIN goroutine stack dump ===\n%s\n=== END goroutine stack dump ===", buf)
+
+	// Also write to file to aid gathering diagnostics
+	name := filepath.Join(os.TempDir(), fmt.Sprintf("containerd.%d.stacks.log", os.Getpid()))
```

Comment:
Can we log the location where the dump is written to?

Reply:
Also wondering if we should add something random (or a time stamp) to the name, as Pid might be reused

Reply:
Any reason not to just use https://golang.org/pkg/io/ioutil/#TempFile . That adds the random component and does the create.

Reply:
@thaJeztah @dmcgowan Randomising it is actually the complete opposite of what I was deliberately trying to achieve....

Reply:
I should probably explain the rationale here - getting stacks out is really a pretty advanced thing to do. You really only care about the stacks at this point in time, usually to investigate a deadlock.  For that reason, in many ways even using the PID here is overkill.